README.md

```diff
@@ -1,2 +1,39 @@
 # rickshaw
 A project to facilitate execution of benchmarks and tools via extensions for many target environments
+
+### Input
+
+Rickshaw needs the following:
+- Benchmark config file
+  - Rickshaw really does not care what the benchmark is; however, rickshaw must have a config file to know how to run the benchmark.  These config files are not bundled with the rickshaw project, but some from other benchmark-helper projects, or you write this config file yourself for your own benchmark.  The config file instructs rickshaw on what to execute on the controller (the host where you are running rickshaw as well as what to run on any benchmark clients and servers. 
+- Benchmark parameters
+  - This may be an array of multiple parameters, when the user wants the benchmark run mulitple ways, for example:
+    <pre>[ "--rw=read --bs=4k", "--rw=write --bs=4k" ]</pre>
+  - The [multiplex](https://github.com/perftool-incubator/multiplex) project can be used to generate this array (it can convert things like "--rw=read,write --bs=4k" into the array above), and it will do parameter validation for you as well.
+- Endpoint extension
```

Comment:
While I know what you are talking about, it probably would be helpful to include a definition of what an endpoint is.  I have a sneaking suspicion that many wouldn't understand that.